libtiff/libtiff_ctypes.py

```diff
@@ -524,24 +524,31 @@ def get_tag_define(tagname):
     @classmethod
     def open(cls, filename, mode='r'):
         """ Open tiff file as TIFF.
+
+        Parameters
+        ----------
+        filename: str or bytes.
```

Comment:
Can we stick to `str` only? That is, in `open`, the filename needs to be encoded to `bytes` but is there are use case where callers of `open` would prefer giving the `filename` as `bytes` object?

Reply:
In our case, we always use `str` to encode the path, as a convention in our application. However, so far in Python the convention is to accept both str and bytes for paths (like the argument to `open()`). It's called a "path-like object": https://docs.python.org/3/glossary.html#term-path-like-object
So I'd suggest to follow the same convention, and give freedom to the user.

Reply:
I've now changed the code to call `os.fspath()`, so that it should work with any path-like object (eg, also the PathLib).